code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -91,12 +91,27 @@ var/const/INGEST = 2
 
 	return the_id
 
-/datum/reagents/proc/trans_to(var/obj/target, var/amount=1, var/multiplier=1, var/preserve_data=1)//if preserve_data=0, the reagents data will be lost. Usefull if you use data for some strange stuff and don't want it to be transferred.
-	if (!target )
+/datum/reagents/proc/trans_to(var/target, var/amount=1, var/multiplier=1, var/preserve_data=1)//if preserve_data=0, the reagents data will be lost. Usefull if you use data for some strange stuff and don't want it to be transferred.
+	if (!target)
 		return
-	if (!target.reagents || src.total_volume<=0)

```

Comment:
After it being explained to me better how this could be used, I understand the point of this and retract my suggestion. (For those who weren't involved: mob.vessel)